common/internal-types/src/channels.rs

```diff
@@ -50,6 +51,19 @@ impl PartialEq for ChannelStatus {
 }
 impl Eq for ChannelStatus {}
 
+impl FromStr for ChannelStatus {
+    type Err = GeneralError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Closed" => Ok(ChannelStatus::Closed),
+            "Open" => Ok(ChannelStatus::Open),
+            "PendingToClose" => Ok(ChannelStatus::PendingToClose(SystemTime::now())),
```

Comment:
This seems weird - if I deserialize from string, I get an instance of `PendingToClose` with no grace period. 

Do we even need this string conversion of ChannelStatus ?

Reply:
You could use the abilities of the `DisplayFromStr` and `DurationSeconds` from `serde_with`.

Reply:
I would be in favor of creating 
a new discriminant enum (https://docs.rs/strum/latest/strum/derive.EnumDiscriminants.html), have that implement `ToString` and `FromStr` (also via appropriate Strum derives) and use that in the REST API instead, if possible.